src/lib.rs

```diff
@@ -424,6 +424,34 @@ impl<S> StreamUnordered<S> {
         })
     }
 
+    /// Returns an immutable iterator that allows getting a reference to each stream in the set.
+    pub fn iter(&self) -> Iter<'_, S>
+    where
+        S: Unpin,
+    {
+        Iter(Pin::new(self).iter_pin())
+    }
+
+    /// Returns an immutable iterator that allows getting a reference to each stream in the set.
+    pub fn iter_pin(self: Pin<&Self>) -> IterPinRef<'_, S> {
```

Comment:
I think we should just not expose this right now unless we have a concrete use-case that benefits from the pinning in the `&` case.